python/pylibcugraph/pylibcugraph/graphs.pyx

```diff
@@ -0,0 +1,169 @@
+
+    Parameters
+    ----------
+    resource_handle : ResourceHandle
+        Handle to the underlying device resources needed for referencing data
+        and running algorithms.
+
+    graph_properties : GraphProperties
+        Object defining intended properties for the graph.
+
+    src_array : device array type
+        Device array containing the vertex identifiers of the source of each
+        directed edge. The order of the array corresponds to the ordering of the
+        dst_array, where the ith item in src_array and the ith item in dst_array
+        define the ith edge of the graph.
+
+    dst_array : device array type
+        Device array containing the vertex identifiers of the destination of each
+        directed edge. The order of the array corresponds to the ordering of the
+        src_array, where the ith item in src_array and the ith item in dst_array
+        define the ith edge of the graph.
+
+    weight_array : device array type
+        Device array containing the weight values of each directed edge. The
+        order of the array corresponds to the ordering of the src_array and
+        dst_array arrays, where the ith item in weight_array is the weight value
+        of the ith edge of the graph.
+
+    store_transposed : bool
+        Set to True if the graph should be transposed. This is required for some
+        algorithms, such as pagerank.
+
+    renumber : bool
+        Set to True to indicate the vertices used in src_array and dst_array are
+        not appropriate for use as internal array indices, and should be mapped
+        to continuous integrers starting from 0.
+
+    do_expensive_check : bool
+        If True, performs more extensive tests on the inputs to ensure
+        validitity, at the expense of increased run time.
+    """
+    def __cinit__(self,
+                  EXPERIMENTAL__ResourceHandle resource_handle,
+                  EXPERIMENTAL__GraphProperties graph_properties,
+                  src_array,
+                  dst_array,
+                  weight_array,
+                  store_transposed,
+                  renumber,
+                  expensive_check):
+
+        # FIXME: add tests for these
+        if not(isinstance(store_transposed, (int, bool))):
+            raise TypeError("expected int or bool for store_transposed, got "
+                            f"{type(store_transposed)}")
+        if not(isinstance(renumber, (int, bool))):
+            raise TypeError("expected int or bool for renumber, got "
+                            f"{type(renumber)}")
+        if not(isinstance(expensive_check, (int, bool))):
+            raise TypeError("expected int or bool for expensive_check, got "
+                            f"{type(expensive_check)}")
+        assert_CAI_type(src_array, "src_array")
+        assert_CAI_type(dst_array, "dst_array")
+        assert_CAI_type(weight_array, "weight_array")
```

Comment:
Should there be an assertion that src_array and dst_array have the same type?  FIXME would be fine for now.

Reply:
added FIXME in next commit.